libs/stdlib/iec61131-st/to_bit.st

```diff
@@ -0,0 +1,246 @@
+FUNCTION TO_BOOL<T: ANY> : BOOL
+    VAR_INPUT
+        in : T;
+    END_VAR
+END_FUNCTION
+
+FUNCTION TO_BOOL__BYTE : BOOL
+    VAR_INPUT
+        in : BYTE;
+    END_VAR
+
+    TO_BOOL__BYTE := BYTE_TO_BOOL(in);
+END_FUNCTION
+
+FUNCTION TO_BOOL__WORD : BOOL
+    VAR_INPUT
+        in : WORD;
+    END_VAR
+
+    TO_BOOL__WORD := WORD_TO_BOOL(in);
+END_FUNCTION
+
+FUNCTION TO_BOOL__DWORD : BOOL
+    VAR_INPUT
+        in : DWORD;
+    END_VAR
+
+    TO_BOOL__DWORD := DWORD_TO_BOOL(in);
+END_FUNCTION
+
+FUNCTION TO_BOOL__LWORD : BOOL
+    VAR_INPUT
+        in : LWORD;
+    END_VAR
+
+    TO_BOOL__LWORD := LWORD_TO_BOOL(in);
+END_FUNCTION
+
+// ======================================================================================================== //
+
+FUNCTION TO_BYTE<T: ANY> : BYTE
+    VAR_INPUT
+        in : T;
+    END_VAR
+END_FUNCTION
+
+FUNCTION TO_BYTE__BOOL : BYTE
+    VAR_INPUT
+        in : BOOL;
+    END_VAR
+
+    TO_BYTE__BOOL := BOOL_TO_BYTE(in);
+END_FUNCTION
+
+FUNCTION TO_BYTE__CHAR : BYTE
+    VAR_INPUT
+        in : CHAR;
+    END_VAR
+
+    TO_BYTE__CHAR := CHAR_TO_BYTE(in);
+END_FUNCTION
+
+FUNCTION TO_BYTE__WORD : BYTE
+    VAR_INPUT
+        in : WORD;
+    END_VAR
+
+    TO_BYTE__WORD := WORD_TO_BYTE(in);
+END_FUNCTION
+
+FUNCTION TO_BYTE__DWORD : BYTE
+    VAR_INPUT
+        in : DWORD;
+    END_VAR
+
+    TO_BYTE__DWORD := DWORD_TO_BYTE(in);
+END_FUNCTION
+
+FUNCTION TO_BYTE__LWORD : BYTE
+    VAR_INPUT
+        in : LWORD;
+    END_VAR
+
+    TO_BYTE__LWORD := LWORD_TO_BYTE(in);
+END_FUNCTION
+
+// ======================================================================================================== //
+
+FUNCTION TO_WORD<T: ANY> : WORD
+    VAR_INPUT
+        in : T;
+    END_VAR
+END_FUNCTION
+
+FUNCTION TO_WORD__BOOL : WORD
+    VAR_INPUT
+        in : BOOL;
+    END_VAR
+
+    TO_WORD__BOOL := BOOL_TO_WORD(in);
+END_FUNCTION
+
+FUNCTION TO_WORD__CHAR : WORD
+    VAR_INPUT
+        in : CHAR;
+    END_VAR
+
+    TO_WORD__CHAR := CHAR_TO_WORD(in);
+END_FUNCTION
+
+FUNCTION TO_WORD__BYTE : WORD
+    VAR_INPUT
+        in : BYTE;
+    END_VAR
+
+    TO_WORD__BYTE := BYTE_TO_WORD(in);
+END_FUNCTION
```

Comment:
I  think you're missing `TO_WORD__WCHAR` here

Reply:
Also, what happens if I call `TO_WORD(WORD#...)` ? The signature seems to be missing also. I am guessing it is gonna default to `BYTE` and truncate.

I think the same-type functions are missing in general. I know it doesn't really make sense to call them, but we should cover it regardless I think - better to err on the safe side.